overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/ignition-ostree-mount-state-overlays.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -euo pipefail
+
+fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+if [ $# -ne 1 ] || { [[ $1 != mount ]] && [[ $1 != umount ]]; }; then
+    fatal "Usage: $0 <mount|umount>"
+fi
+
+# if state overlays are disabled, there's nothing to do
+if ! jq -e '.["opt-usrlocal-overlays"]' /sysroot/usr/share/rpm-ostree/treefile.json; then
+    exit 0
+fi
+
+do_mount() {
+    for overlay in /usr/lib/opt /usr/local; do
+        escaped=$(systemd-escape --path "${overlay}")
+        overlay_dirs=/sysroot/var/ostree/state-overlays/${escaped}
```

Comment:
Since we only operate on one overlay dir at a time maybe:

```suggestion
        overlay_dir=/sysroot/var/ostree/state-overlays/${escaped}
```

Reply:
It's plural because that dir contains `work` and `upper`. This matches the wording in e.g. https://github.com/ostreedev/ostree/blob/da892140657bfb67d475f80395af97af96b1b6af/src/ostree/ot-admin-builtin-state-overlay.c#L48.